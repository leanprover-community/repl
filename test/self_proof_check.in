{"cmd": "theorem ex : False := by exact?"}

{"cmd": "theorem ex : False := sorry", "sorries": "individual"}

{"proofState": 0, "tactic": "exact?"}

{"cmd": "theorem ex : False := by sorry", "sorries": "individual"}

{"proofState": 1, "tactic": "exact?"}

{"cmd": "theorem ex : False := by sorry", "allTactics": true, "sorries": "individual"}

{"proofState": 3, "tactic": "exact?"}

{"cmd": "theorem ex : False := by sorry", "sorries": "rootGoals"}

{"proofState": 4, "tactic": "exact?"}

