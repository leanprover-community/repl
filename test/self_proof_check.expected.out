{"messages":
 [{"severity": "error",
   "pos": {"line": 1, "column": 25},
   "endPos": {"line": 1, "column": 31},
   "data":
   "`exact?` could not close the goal. Try `apply?` to see partial suggestions."}],
 "env": 0}

{"sorries":
 [{"proofState": 0,
   "pos": {"line": 1, "column": 22},
   "parentDecl": "ex",
   "goals": ["⊢ False"],
   "endPos": {"line": 1, "column": 27}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 8},
   "endPos": {"line": 1, "column": 10},
   "data": "declaration uses 'sorry'"}],
 "env": 1}

{"message":
 "Lean error:\n`exact?` could not close the goal. Try `apply?` to see partial suggestions."}

{"sorries":
 [{"proofState": 1,
   "pos": {"line": 1, "column": 25},
   "parentDecl": "ex",
   "goals": ["⊢ False"],
   "endPos": {"line": 1, "column": 30}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 8},
   "endPos": {"line": 1, "column": 10},
   "data": "declaration uses 'sorry'"}],
 "env": 2}

{"message":
 "Lean error:\n`exact?` could not close the goal. Try `apply?` to see partial suggestions."}

{"tactics":
 [{"usedConstants":
   ["False",
    "sorryAx",
    "instOfNatNat",
    "Lean.Name.num",
    "Lean.Name.str",
    "Lean.Name.anonymous",
    "Nat",
    "Lean.Name",
    "OfNat.ofNat",
    "Bool.false"],
   "tactic": "sorry",
   "proofState": 3,
   "pos": {"line": 1, "column": 25},
   "parentDecl": "ex",
   "goals": "⊢ False",
   "endPos": {"line": 1, "column": 30}}],
 "sorries":
 [{"proofState": 2,
   "pos": {"line": 1, "column": 25},
   "parentDecl": "ex",
   "goals": ["⊢ False"],
   "endPos": {"line": 1, "column": 30}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 8},
   "endPos": {"line": 1, "column": 10},
   "data": "declaration uses 'sorry'"}],
 "env": 3}

{"message":
 "Lean error:\n`exact?` could not close the goal. Try `apply?` to see partial suggestions."}

{"sorries":
 [{"proofState": 4,
   "pos": {"line": 1, "column": 22},
   "parentDecl": "ex",
   "goals": ["⊢ False"]}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 8},
   "endPos": {"line": 1, "column": 10},
   "data": "declaration uses 'sorry'"}],
 "env": 4}

{"message":
 "Lean error:\n`exact?` could not close the goal. Try `apply?` to see partial suggestions."}

